src/Tickets/Commerce/Traits/Type.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Type trait.
+ *
+ * @since TBD
+ */
+
+namespace TEC\Tickets\Commerce\Traits;
+
+/**
+ * Trait Type
+ *
+ * @since TBD
+ */
+trait Type {
+
+	use Is_Ticket;
+
+	/**
+	 * Determine if a thing is a coupon.
+	 *
+	 * This looks to see whether the array of data has the "type" key set to
+	 * "coupon". If the type key is not set, or if it is set to something other
+	 * than "coupon", this will return false.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $thing The thing to check.
+	 *
+	 * @return bool Whether the thing is a coupon.
+	 */
+	protected function is_coupon( array $thing ): bool {
```

Comment:
Why do we need this _and_ the one in Coupon above? Too much specificity here?

Reply:
They do different things, but that's a good point. There should only be 1.